plugin/keyring_vault/vault_curl.h

```diff
@@ -38,15 +37,14 @@ class Vault_curl : public IVault_curl, private boost::noncopyable
 
 private:
 
-  bool reset_curl_session();
+  bool setup_curl_session(CURL *curl);
   std::string get_error_from_curl(CURLcode curl_code);
   bool encode_key_signature(const Vault_key &key, Secure_string *encoded_key_signature);
   bool get_key_url(const Vault_key &key, Secure_string *key_url);
 
   ILogger *logger;
   Secure_string token_header;
   Secure_string vault_url;
-  CURL *curl;
```

Comment:
Were there any problems with keeping `curl` as a class member and calling `curl_easy_reset(curl)` before each request?

Reply:
Yes. For some reason CURL is not coping very well with repetitive new connections using the same CURL object. There were quite a few "CURL not initialized correctly" errors in error logs - which by its own should not be a problem - CURL should just stop working, but under stress testing it caused CURL to crash. I think it should be safer this way to just use fresh CURL object per each operation than make yet another try to reuse CURL object in a way that will not cause a crash. The reason is also that Centos6 is using very old CURL - from 2008 and it was a source of lots of troubles and I do not think it is a reliable library.